### PR TITLE
begin release procedure

### PR DESCRIPTION
**Pull request by Atticus29**
Pull request checklist:

- [ ] Features and bugfixes should be PRed into the `develop` branch, **not** `main`
    - NA
- [ ] All text is internationalized
    - NA
- [ ] There are no linter errors
    NA
- [ ] New features support all states (loading, error, etc)
    NA

Which JIRA ticket(s) and/or GitHub issues does this PR address?
NA.

Thanks for keeping it clean! Feel free to merge your own pull request after it has been approved - just use the "squash & merge" option.

### CODE REVIEW

File: docs/release-procedure.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+1. `git checkout develop`
+2. `git pull origin develop`
+3. `git checkout main`
+4. `git merge develop` (there should be no merge conflicts)
+
+Then, draft a new release of the main branch on GitHub:
+
+1. Navigate to [https://github.com/WildMeOrg/codex-frontend/releases/new](https://github.com/WildMeOrg/codex-frontend/releases/new).
+2. Designate the target as the main branch.
+3. Create a new tag following the SemVer pattern outlined [here](https://semver.org/) (vX.X.X).
```
Comment (brmscheiner) [inline docs/release-procedure.md:14]: make it more explicit that we need to have a string that looks exactly like vX.X.X - i know you say that here but i want to be very clear about it, since devOps services will be looking for that exact string

Comment (brmscheiner) [reply docs/release-procedure.md:14]: also the tag should only be made after the PR merging develop into main has been approved and merged

File: docs/release-procedure.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# Release procedure
```
Comment (brmscheiner) [inline docs/release-procedure.md:1]: link to this in the contribution guide please

Comment (Atticus29) [reply docs/release-procedure.md:1]: Update in https://github.com/WildMeOrg/codex-frontend/pull/424/commits/5f52be6481f1d4a5568fa496ff399b5d68ad7346

File: docs/release-procedure.md (hunk 1 above)
Comment (Atticus29) [reply docs/release-procedure.md]: Updated in https://github.com/WildMeOrg/codex-frontend/pull/424/commits/5f52be6481f1d4a5568fa496ff399b5d68ad7346.

File: docs/release-procedure.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+First, resolve any potential merge conflicts with main on the develop branch
+
+1. `git checkout develop`
+2. `git pull origin develop`
+3. `git checkout main`
+4. `git merge develop` (there should be no merge conflicts)
+
```
Comment (Emily-Ke) [inline docs/release-procedure.md:9]: Then create a PR into develop to update the version.

File: docs/release-procedure.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
+# Release procedure
```
Comment (Emily-Ke) [inline docs/release-procedure.md:1]: A release PR should not happen while there are remaining work items in QA. What if there are items in QA, or other previous merges that we want to omit from the release?

`develop` should be frozen during this time.

Comment (Atticus29) [reply docs/release-procedure.md:1]: What do you think about: https://github.com/WildMeOrg/codex-frontend/pull/424/commits/75c6e6145959b784474ec65055f9b13667dd89c1?

File: docs/release-procedure.md [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+The develop branch should be "frozen" during the release procedure; please be sure that everyone on the team is notified of the freeze and when it is completed.
+
+## Merge and create the PR
+
+First, resolve any potential merge conflicts with main on the develop branch
+
+1. `git checkout develop`
+2. `git pull origin develop`
+3. `git checkout main`
+4. `git merge develop` (there should be no merge conflicts)
```
Comment (Emily-Ke) [inline docs/release-procedure.md:18]: `main` should also be up to date.
```
3. `git checkout main`
4. `git pull origin main`
5. `git merge develop` (there should be no merge conflicts)
```

Comment (Emily-Ke) [inline docs/release-procedure.md:18]: >(there should be no merge conflicts)

But if there are, then what? The conflicts need to be resolved in a separate PR into `develop`.

File: docs/release-procedure.md [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+5. Update the version in package.json.
+
+Then, issue a pull request for merging the develop branch into the main branch. Await approval and then merge.
```
Comment (Emily-Ke) [inline docs/release-procedure.md:21]: We're not using the "Squash and Merge" method for this are we? If not, I think that we should be explicit here since it is different than the behavior for our other pull requests?

File: docs/release-procedure.md [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+
+## Draft a new release
+
+After completion of the previous steps, draft a new release of the main branch on GitHub:
+
+1. Navigate to [https://github.com/WildMeOrg/codex-frontend/releases/new](https://github.com/WildMeOrg/codex-frontend/releases/new).
+2. Designate the target as the main branch.
+3. Create a new tag following the SemVer pattern outlined [here](https://semver.org/) (vX.X.X). Note that this exact pattern (no period between v and the first name, for instance) must be followed explicitly.
```
Comment (Emily-Ke) [inline docs/release-procedure.md:29]: The release title should be the same as the tag.

Comment (Emily-Ke) [inline docs/release-procedure.md:29]: Do we want consistent guidelines for the release description?

Comment (Emily-Ke) [inline docs/release-procedure.md:29]: This should possibly be (vX.Y.Z)

File: docs/release-procedure.md (hunk 4 above)
Comment (Emily-Ke) [reply docs/release-procedure.md:1]: I was hoping for actual procedures if some commits in `develop` need to be omitted. @brmscheiner, what are your thoughts on this?

File: docs/release-procedure.md (hunk 6 above)
Comment (brmscheiner) [reply docs/release-procedure.md:21]: I think squash and merge would be fine. Develop would still have the full commit history but `main` would have a much cleaner history

**Review by brmscheiner — APPROVED** [review]

**Review by Emily-Ke — APPROVED** [review]
